data_search/analytical_data_search.md

```diff
@@ -0,0 +1,1144 @@
+print(tmc1a_jwst_obstable[0]['dataURL'])
+```
+
+So we can save ourselves quite a lot of time by feeding a python list of these `dataURL`s directly into `astroquery.Observations.get_cloud_uris`, which will convert each `dataURL` into a cloud URI.  By setting `return_uri_map=True`, we'll also ensure that we know which observation each cloud URI corresponds to. The output `cloud_uri_map` is a python dictionary mapping `dataURL` values to cloud URIs.
+
+We'll also set `verbose=False` to suppress warnings about data products that are temporarily unavailable in the cloud; be sure to turn `verbose` back on if your science case requires complete information, or add code later on to handle the situation where a cloud URI is not available.
+
+```{code-cell} ipython3
+cloud_uri_map = Observations.get_cloud_uris(list(tmc1a_jwst_obstable['dataURL']), return_uri_map=True, verbose=False)
+```
+
+We could just look at the first observation in the table, but for demonstration purposes and to ensure that the images we plot in this tutorial are predictable, let's retrieve the observation corresponding to the first row in this table meeting the following criteria: MIRI instrument, CH1-SHORT dispersion element, proposal ID 1290, and has a valid cloud URI.
+
+```{code-cell} ipython3
+# Retrieve all observations for TMC1A with MIRI CH1-SHORT and proposal ID 1290
+temp_observations = tmc1a_jwst_obstable[
+    (tmc1a_jwst_obstable['instrument_name']=='MIRI/IFU') &
+    (tmc1a_jwst_obstable['filters']=='CH1-SHORT') &
+    (tmc1a_jwst_obstable['proposal_id']=='1290')
+]
+
+# Get the first valid cloud URI found
+for row in temp_observations:
+    cloud_uri = cloud_uri_map[row['dataURL']]
+    if cloud_uri:
+        break  # When a valid URI is found, exit the loop
+
+print('Cloud URI: ', cloud_uri)
+```
+
+MAST cloud data is in the [Registry of Open Data on AWS](https://registry.opendata.aws/), so it's configured to allow non-credentialed anonymous access. We can load this file into `astropy.fits` by passing in the cloud URI in as if it were a normal URL, but we need to tell `astropy.fits` to access the file anonymously. More information is available [in the astropy documentation](https://docs.astropy.org/en/stable/io/fits/usage/cloud.html).
+
+```{code-cell} ipython3
+with fits.open(cloud_uri, cache=False, use_fsspec=True, fsspec_kwargs={"anon": True}) as hdul:
+    header = hdul['SCI'].header
+    data = hdul['SCI'].data
+
+# Note: we didn't actually need to manually set use_fsspec=True.
+# This is turned on by default when using an AWS URI.
+
+# Note: instead of using fsspec inside an astropy.fits.open() to handle the cloud connection,
+# you could equivalently use the s3fs module explicitly, as follows.
+# Results are typically similar.
+
+# import s3fs
+# fs = s3fs.S3FileSystem(anon=True)
+# with fs.open(cloud_uri, 'rb') as cloud_file:
+#     with fits.open(cloud_file, 'readonly') as hdul:
+#         header = hdul['SCI'].header
+#         data = hdul['SCI'].data
```

Comment:
Please remove, no need for this commentary, we do not go one layer deep and explain all astropy functionality.